pandas/io/excel/_util.py

```diff
@@ -23,32 +23,46 @@ def register_writer(klass):
     _writers[engine_name] = klass
 
 
-def get_default_writer(ext):
+def get_default_engine(ext, mode="read"):
     """
-    Return the default writer for the given extension.
+    Return the default reader/writer for the given extension.
 
     Parameters
     ----------
     ext : str
         The excel file extension for which to get the default engine.
+    mode : str
+        Whether to get the default engine for reading or writing.
+        Either 'read' or 'write'
 
     Returns
     -------
     str
         The default engine for the extension.
     """
+    _default_readers = {
+        "xlsx": "openpyxl",
+        "xlsm": "openpyxl",
+        "xlsb": "pyxlsb",
+        "xls": "xlrd",
+        "ods": "odf",
+    }
     _default_writers = {
         "xlsx": "openpyxl",
         "xlsm": "openpyxl",
+        "xlsb": "pyxlsb",
         "xls": "xlwt",
         "ods": "odf",
     }
-    xlsxwriter = import_optional_dependency(
-        "xlsxwriter", raise_on_missing=False, on_version="warn"
-    )
-    if xlsxwriter:
-        _default_writers["xlsx"] = "xlsxwriter"
-    return _default_writers[ext]
+    if mode == "write":
+        xlsxwriter = import_optional_dependency(
```

Comment:
why is this checked here cc @rhshadrach this is just a mapping right? when the engine is constructed this should actually be checked.

Reply:
IOW get_default_engine should never raise (or if we choose to do this, then it should be documented, and if this is the case). why are we not importing openpyxl for readers here? its just pretty inconsistent.

Reply:
This doesn't raise. Rather, it changes the default engine based on what the user has installed. So the logic for xlsx files is "if user has xlsxwriter installed, use it as the default; otherwise use openpyxl".

Reply:
ahh ok makes sense, ok then

@lithomas1 can you add a commen to this effect here

Reply:
Yeah, this is old code, from when the function was get_default_writers. I'll add in the comment.